pkg/gzip/gzip.go

```diff
@@ -69,7 +77,29 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 			)
 		}
 
+		existing := firstExistingParent(fpath)
+		if realParent, err := filepath.EvalSymlinks(existing); err == nil {
+			if !strings.HasPrefix(realParent, realDest) {
```

Comment:
the test above: `!strings.HasPrefix(fpath, dest)` is subsumed by this, so it could be deleted. the only difference is the error message (the error above doesn't mention symlinks, this error does). think I would like to present a different error message in the two cases, so maybe I'll leave it.